app/controllers/works_controller.rb

```diff
@@ -0,0 +1,91 @@
+class WorksController < ApplicationController
+  before_action :works_all, only: [:index, :main]
+  before_action :find_work, only: [:show, :edit, :update]
+
+  def index; end
+
+  def main; end
```

Comment:
I think I'd prefer to see the instance variables set here to pull each work type rather than these pulled directly from the view.  That way, if the model methods ever changed, the view would remain the same and only the controller action would change.